### PR TITLE
Vertical blind warning not reported from material get input function

### LINKED ISSUE #10742: Vertical blinds do not report a warning when slat curvature is < slat width / 2
Issue overview
--------------
A logic mistake:

        if ((mat->LayerType == TARCOGParams::TARCOGLayerType::VENETBLIND_HORIZ) ||
            (mat->LayerType == TARCOGParams::TARCOGLayerType::VENETBLIND_HORIZ)) {

hides a warning for vertical blinds. An error is reported by tarcog but not by the material getInput function.

    ** Severe  ** Window tarcog returned an error
    **   ~~~   ** message = "Invalid curvature radius (absolute value must be >SlatWidth/2, or 0 for flat slats). Layer #  2"
    **   ~~~   **  During Warmup & Sizing, Environment=CHICAGO OHARE INTL AP ANN HTG 99.6% CONDNS DB, at Simulation time=01/21 00:00 - 00:00
    **   ~~~   ** construction name = CFS_GLZ_1
    **  Fatal  ** halting because of error in tarcog

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added ([CmplxGlz_Daylighting_SouthVerticalVB45deg-in.txt](https://github.com/user-attachments/files/17007966/CmplxGlz_Daylighting_SouthVerticalVB45deg-in.txt))

 - [ ] Ticket added to EnergyPlus Defect Complexity (Github Project)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes #10742
 - Material input for WindowMaterial:ComplexShade was not reporting an error for blind slat curvature while that error was reported from the Tarcog module TARCOGArgs.cc.
 
**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/Material.cc [hunk 1]
```diff
@@ -2668,45 +2668,12 @@ void GetMaterialData(EnergyPlusData &state, bool &ErrorsFound) // set to true if
...
-            if (s_ipsc->rNumericArgs(15) <= 0.0) {
-                ErrorsFound = true;
-                ShowSevereCustomMessage(
-                    state, eoh, format("{} must be >0, entered value = {:.2R}", s_ipsc->cNumericFieldNames(15), s_ipsc->rNumericArgs(15)));
-            }
-
-            if ((s_ipsc->rNumericArgs(16) < 0.0) ||
-                ((s_ipsc->rNumericArgs(16) > 0.0) && (s_ipsc->rNumericArgs(16) < (s_ipsc->rNumericArgs(11) / 2)))) {
+            (mat->LayerType == TARCOGParams::TARCOGLayerType::VENETBLIND_VERT)) {
+            if ((s_ipsc->rNumericArgs(16) > 0.0) && (s_ipsc->rNumericArgs(16) < (s_ipsc->rNumericArgs(11) / 2))) {
```
Comment (rraustad) [inline src/EnergyPlus/Material.cc:2672]: [src/EnergyPlus/Material.cc:2670]:(style),[duplicateExpression],Same expression on both sides of '||'.

These other checks are caught by InputProcessor and are never executed. For example if I make slat thickness < 0 this input processor warning is reported.

    ** Severe  ** <root>[WindowMaterial:ComplexShade][Shade_14_Layer][slat_thickness] - "-0.001000" - Expected number greater than 0.000000
    **  Fatal  ** Errors occurred on processing input file. Preceding condition(s) cause termination.

    WindowMaterial:ComplexShade,

     N11 , \field Slat Width
         \units m
         \type real
         \minimum> 0
         \default 0.016
     N12 , \field Slat Spacing
         \note Distance between adjacent slat faces
         \units m
         \type real
         \minimum> 0
         \default 0.012
     N13 , \field Slat Thickness
         \note Distance between top and bottom surfaces of slat
         \note Slat is assumed to be rectangular in cross section and flat
         \type real
         \units m
         \minimum> 0
         \default 0.0006
     N14 , \field Slat Angle
         \type real
         \units deg
         \default 90
         \minimum -90
         \maximum 90
     N15 , \field Slat Conductivity
         \type real
         \units W/m-K
         \minimum> 0
         \default 160.0
     N16 ; \field Slat Curve
         \note this value represents curvature radius of the slat.
         \note if the slat is flat use zero.
         \note if this value is not zero, then it must be > SlatWidth/2.
         \type real
         \units m
         \minimum 0.0
         \default 0

File: src/EnergyPlus/Material.cc [hunk 2]
```diff
@@ -2496,14 +2496,14 @@ void GetMaterialData(EnergyPlusData &state, bool &ErrorsFound) // set to true if
...
+            auto const &instances2 = itInstances.value();
+            auto itObj = instances2.begin();
             // Can't use find here because epJSON keys are not upper-cased
-            for (; itObj != instances.end(); ++itObj) {
+            for (; itObj != instances2.end(); ++itObj) {
                 if (Util::makeUPPER(itObj.key()) == s_ipsc->cAlphaArgs(3)) break;
             }
 
-            if (itObj == instances.end()) {
+            if (itObj == instances2.end()) {
```
Comment (rraustad) [inline src/EnergyPlus/Material.cc:2506]: [src/EnergyPlus/Material.cc:2499]:(style),[shadowVariable],Local variable 'instances' shadows outer variable
[src/EnergyPlus/Material.cc:2526]:(style),[shadowVariable],Local variable 'instances' shadows outer variable

File: src/EnergyPlus/Material.cc [hunk 3]
```diff
@@ -84,7 +84,7 @@ constexpr std::array<Material::Gas, 10> gases = {
 
 constexpr std::array<std::string_view, (int)EcoRoofCalcMethod::Num> ecoRoofCalcMethodNamesUC = {"SIMPLE", "ADVANCED"};
 
-int GetMaterialNum(EnergyPlusData &state, std::string const &matName)
+int GetMaterialNum(EnergyPlusData const &state, std::string const &matName)
```
Comment (rraustad) [inline src/EnergyPlus/Material.cc:87]: [src/EnergyPlus/Material.cc:87]:(style),[constParameter],Parameter 'state' can be declared as reference to const

**Comment by rraustad** [comment]
Defect file with this branch:

    ** Severe  ** GetMaterialData: WindowMaterial:ComplexShade = SHADE_14_LAYER
    **   ~~~   ** Slat Curve must be = 0 or greater than SlatWidth/2, entered value = 2.00E-003
    **  Fatal  ** Error in complex fenestration material input.

File: src/EnergyPlus/Material.cc (hunk 1 above)
Comment (Myoldmopar) [reply src/EnergyPlus/Material.cc:2672]: Yes!  There is so much duplicated between the IDD/input processor range checks and then inaccessible checks in the get input routines.  I love it.

**Review by Myoldmopar — APPROVED** [review]
These changes look fine.  But I am a little confused.  The title is 

> Vertical blind warning not reported from material get input function

But I don't really see how your changes here enabled a warning to be emitted.  Was it the `instances` variable shadowing?

**Comment by rraustad** [comment]
For the warning, a copy paste error caused the vertical blind to be skipped.

image

and now it's corrected here:

image

For the shadow variable there is the same variable outside this scope up at 177. I wasn't sure if I could reuse that one so I just renamed it.

image

**Comment by Myoldmopar** [comment]
Ahh I see.  Thanks!  Windows build failure is certainly unrelated, right?  But do I get fresh results just to be sure?  Ugh, probably.  OK, I'll merge develop in and get fresh results.

**Comment by Myoldmopar** [comment]
There we go, all clean.  Thanks @rraustad, merging.

**Pull request merged**